Automattic/wp-calypso
# Jetpack Manage Pricing page: Refactor redirect logic so it works with authorizePath()

Resolves https://github.com/Automattic/jetpack-manage/issues/279

If a non-logged in user clicks on any of the `Get` buttons on `/manage/pricing` they'd be correctly redirected to the "login/auth" part of Calypso.

Before this PR, `window.location` would end up being set to the origin page by the `authorizePath()` which meant users would be sent back to `/manage/pricing` instead of to `/partner-portal/issue-license`

## Proposed Changes

* Replace `href` properties with `page.redirect()` functions

## Testing Instructions



* Go to `/manage/pricing` without being logged in
* Press any "Get" buttons on the page
* Login to your WPCOM account
* Authorize that Jetpack Cloud can use your WPCOM account
* Verify that you end up on `/partner-portal/issue-license` and that it shows a modal that looks like the below screenshot

![Screenshot 2024-02-05 at 23 14 54](https://github.com/Automattic/wp-calypso/assets/3846700/c2404582-2cf9-4aeb-af05-e806d8dc6a50)

## Pre-merge Checklist
